### PR TITLE
Add performance test scripts

### PR DESCRIPTION
**Pull request by lipppppp**
Hi, I have changed the directory structure as `ec-performance-test`. Following your advice, I added README.md for each test and deleted the util.py.

### CODE REVIEW

File: supports/compression-performance-test/run_ssm_compress.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+num = sys.argv[2]
+
+#The data dir is named by case. Please see prepare.sh
+case = size + "_" + num
+log = sys.argv[3]
+#Either "ec" or "unec"
+action = sys.argv[4]
+
+if action == "compressDFSIO":
+    rid = submit_rule("file: path matches \"/" + case + "/*\"| compress -compressImpl Zlib")
```
Comment (philo-he) [inline supports/compression-performance-test/run_ssm_compress.py]: In #2107, I had replaced '-compresssImpl' by '-codec' for compression action. Please update the compression rules in line 15 & line 18 accordingly.

File: supports/mover-performance-test/README.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Performance Test for SSM Mover
+Test the performance of SSM Mover and compare it with HDFS Mover. We user mover to migrate data from all_disk to all_ssd. You can also choose other storage policies to test the mover performance.
```
Comment (philo-he) [inline supports/mover-performance-test/README.md]: We use SSM mover & HDFS mover to migrate ...

File: supports/mover-performance-test/README.md [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+## Configuration
+  Configure the file named `config`. For a test case, the corresponding test data should be created in the HDFS cluster beforehand by executing 'prepare.sh'.
+
+## Compression Test
+  1. Run `./test_hdfs_mover_performance.sh` and `./test_ssm_mover_performance.sh`
+  2. A file named ssm.log under this directory will record the time for each round of test. SSM log and PAT data will be collected in `${PAT_HOME}/PAT-collecting-data/results`.
+
+
+## Notes
+- You should configure `dfs.datanode.data.dir` and HDFS HSM related configuration before you run this test. For example, you can set datanode path as `/hadoop/hdfs/data1,[SSD]/hadoop/hdfs/data2`.
```
Comment (philo-he) [inline supports/mover-performance-test/README.md]: For example, you can set the value for `dfs.datanode.data.dir` with `[DISK]/mnt/disk1,[SSD]/mnt/nvme`.

I think explicitly putting a 'DISK' tag would make the example more clear to user.

File: supports/mover-performance-test/config [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+SMART_HOME=
+
+# PAT home, e.g., PAT_HOME=~/PAT
+PAT_HOME=
+
+# Hadoop home, e.g., HADOOP_HOME=~/hadoop
+HADOOP_HOME=
+
+# The cluster hdfs url for ec performance test, e.g., SRC_CLUSTER=hdfs://sr613:9000
+SRC_CLUSTER=
```
Comment (philo-he) [inline supports/mover-performance-test/config]: It looks that 'SRC_CLUSTER', 'DEST_DIR_EC', 'DEST_DIR_REPLICA' & 'MAPPER_NUM' are useless in this test. Please remove them.

File: supports/compression-performance-test/config [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+PAT_HOME=
+
+# Hadoop home, e.g., HADOOP_HOME=~/hadoop
+HADOOP_HOME=
+
+# HiBench home, e.g., HiBench_HOME=~/HiBench
+HiBench_HOME=
+
+# The cluster hdfs url for ec performance test, e.g., SRC_CLUSTER=hdfs://sr613:9000
+SRC_CLUSTER=
```
Comment (philo-he) [inline supports/compression-performance-test/config]: It looks that 'SRC_CLUSTER', 'DEST_DIR_EC', 'DEST_DIR_REPLICA' & 'MAPPER_NUM' are useless in this test. Please remove them.

File: supports/mover-performance-test/test_hdfs_mover_performance.sh [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+echo "Hadoop home is ${HADOOP_HOME}"
+echo "Test case:"
+for size in ${!CASES[@]}; do
+  echo ${size} ${CASES[$size]}
+done
+echo "--------------------------------------------------------"
+
+bin=$(dirname "${BASH_SOURCE-$0}")
+bin=$(cd "${bin}">/dev/null; pwd)
+log="${bin}/ssm.log"
```
Comment (philo-he) [inline supports/mover-performance-test/test_hdfs_mover_performance.sh]: The log can be named as 'hdfs.log', thus, we can distinguish it from SSM log.

File: supports/mover-performance-test/test_ssm_mover_performance.sh [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+	sh drop_cache.sh
+        # make ssm log empty before test
+        printf "" > ${SMART_HOME}/logs/smartserver.log
+        cd ${PAT_HOME}/PAT-collecting-data
+        # hdfs user will execute the cmd, you can change it to the one in your test env. with execution permission
+        echo "export PYTHONPATH=${bin}/../integration-test:${PYTHONPATH};\
+         python ${bin}/run_ssm_mover.py ${size} ${CASES[$size]} ${log} ${action};\
+        su hdfs -c \"hadoop jar ${HADOOP_HOME}/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar \
+        TestDFSIO -Dtest.build.data=/\"${size}_${CASES[$size]}\" \
+        -read -nrFiles ${CASES[$size]} -size ${size} -resFile /home/hdfs/dfsio_mover_test.log\"" > cmd.sh
```
Comment (philo-he) [inline supports/mover-performance-test/test_ssm_mover_performance.sh]: This test is in order to compare the exectution performance of HDFS mover and SSM mover. So there maybe no need to run DFSIO read test. 

I think you can remove the below content and keep `"> cmd.sh` (line 47).

su hdfs -c \"hadoop jar ${HADOOP_HOME}/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar \
        TestDFSIO -Dtest.build.data=/\"${size}_${CASES[$size]}\" \
        -read -nrFiles ${CASES[$size]} -size ${size} -resFile /home/hdfs/dfsio_mover_test.log\"

File: supports/mover-performance-test/test_ssm_mover_performance.sh [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+	action="alldisk"
+	echo "Test case ${case}($action):" >> ${log}
+        echo "==================== test case: $case, test round: $i ============================"
+        sh drop_cache.sh
+        cd ${PAT_HOME}/PAT-collecting-data
+        echo "export PYTHONPATH=${bin}/../integration-test:${PYTHONPATH};\
+         python ${bin}/run_ssm_mover.py ${size} ${CASES[$size]} ${log} ${action};\
+         su hdfs -c \"hadoop jar ${HADOOP_HOME}/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar \
+         TestDFSIO -Dtest.build.data=/\"${size}_${CASES[$size]}\" \
+         -read -nrFiles ${CASES[$size]} -size ${size} -resFile /home/hdfs/dfsio_mover_test.log\"" > cmd.sh
```
Comment (philo-he) [inline supports/mover-performance-test/test_ssm_mover_performance.sh]: Similar to the above, DFSIO read test should be removed.

File: supports/mover-performance-test/prepare.sh [hunk 9]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+
+. ./config
+
+# generate test data using DFSIO
+for size in "${!CASES[@]}"; do
+    num=${CASES[$size]}
+    dir="${size}_${num}"
+    ssh ${SRC_NODE} "hdfs dfs -mkdir -p /${dir}/io_data"
+    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile /home/hdfs/dfsio_mover_test.log
```
Comment (philo-he) [inline supports/mover-performance-test/prepare.sh]: The log path is env. dependent. I recommend to use `$(pwd)/dfsio_mover_test.log`.

File: supports/compression-performance-test/prepare.sh [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+# generate test data using Teragen
+chmod 777 -R ${HiBench_HOME}
+sh ${HiBench_HOME}/bin/workloads/micro/terasort/prepare/prepare.sh
+
+# generate test data using TestDFSIO
+for size in "${!CASES[@]}"; do
+    num=${CASES[$size]}
+    dir="${size}_${num}"
+    ssh ${SRC_NODE} "hdfs dfs -mkdir -p /${dir}/io_data"
+    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile /home/hdfs/dfsio_mover_test.log
```
Comment (philo-he) [inline supports/compression-performance-test/prepare.sh]: The log path is env. dependent. I recommend to use `$(pwd)/dfsio_mover_test.log`.

**Comment by philo-he** [comment]
Thanks @lipppppp for your patch. Please revise the patch again and then directly push the commits to your working branch on github. The pull request will follow the new commits.

**Comment by lipppppp** [comment]
Thank you for your advice. It's really helpful for me. I updated these scripts.

File: supports/compression-performance-test/prepare.sh [hunk 11]
```diff
@@ -11,7 +11,7 @@ for size in "${!CASES[@]}"; do
     num=${CASES[$size]}
     dir="${size}_${num}"
     ssh ${SRC_NODE} "hdfs dfs -mkdir -p /${dir}/io_data"
-    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile /home/hdfs/dfsio_mover_test.log
+    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile ${pwd}/dfsio_compression_test.log
```
Comment (philo-he) [inline supports/compression-performance-test/prepare.sh]: Use $(pwd) instead of ${pwd}.

File: supports/mover-performance-test/prepare.sh [hunk 12]
```diff
@@ -7,7 +7,7 @@ for size in "${!CASES[@]}"; do
     num=${CASES[$size]}
     dir="${size}_${num}"
     ssh ${SRC_NODE} "hdfs dfs -mkdir -p /${dir}/io_data"
-    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile /home/hdfs/dfsio_mover_test.log
+    hadoop jar $HADOOP_HOME/hadoop-mapreduce/hadoop-mapreduce-client-jobclient-*-tests.jar TestDFSIO -write -nrFiles $(($num)) -size ${size}  -resFile ${pwd}/dfsio_mover_test.log
```
Comment (philo-he) [inline supports/mover-performance-test/prepare.sh]: Use $(pwd) instead of ${pwd}.

**Comment by philo-he** [comment]
The patch looks good. Will commit soon. Thanks again!

**Comment by lipppppp** [comment]
Thanks for your review.

**Pull request merged**